web/react/components/modal.jsx

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+export default class Modal extends ReactBootstrap.Modal {
```

Comment:
The parent can check it itself, but I'm trying to avoid needing to duplicate that logic everywhere. Many dialogs use some sort of onShow handler to pull data, set fields, or set the max height, and it's much cleaner to just have these as an event on the Modal like the existing onExited property.

With this, we can just write
```javascript
class ModalA extends React.Component {
    render() {
        return <Modal onShow={this.doStuff} />;
    }
}
```
instead of
```javascript
class ModalB extends React.Component {
    componentDidMount() {
        if (this.props.show) {
            this.doStuff();
        }
    }
    componentDidUpdate(prevProps) {
        if (this.props.show && !prevProps.show) {
            this.doStuff();
        }
    }
}
```
everywhere.

And I needed onPreshow for the EditPostModal at some point, but that modal never got finished so it's currently unused. I'm fine with removing it after I've ported the other modals over, but I'd rather leave it in for now case I find I need it somewhere.